### DISCUSSION TITLE
Is support for namespaces and ClojureScript dependencies a goal?

### DISCUSSION THREAD

**Discussion by martinklepsch** [squint-cljs/cherry#95] [Q&A]
Hello :slightly_smiling_face: I was wondering what the status / goals are with regards to supporting namespaces / other Clojure code (like libraries). Would be curious if you have ideas how that could work and if you see issues because ES6 modules and Clojure namespaces are too different?

I think [this issue](https://github.com/squint-cljs/cherry/issues/64) is also about something similar.

**Comment by borkdude** [comment]
@martinklepsch This is something I'm thinking about but I'm not exactly sure on the details yet.

You can currently embed cherry in an existing CLJS app as follows:

https://github.com/squint-cljs/cherry/blob/main/doc/embed.md

but also in that mode I want to support emitting namespaces.

**Comment by kloimhardt** [comment]
@martinklepsch Although only loosely related to your question, I experimented with the use of `cherry.embed/compile-string` (i.e convert cljs-code -> js-code) in a HTML <script> tag. 

Turns out that in the according Shadow-cljs app, `cherry.embed/compile-string` and `(cherry.embed/preserve-ns 'cljs.core)` are allowed to be in different namespaces. Those can also be code-split into different js-files as described in the shadow-cljs guide. And sometimes `preserve-ns` is not even necessary for `compile-string` to work.

But all that is only heuristic findings through experiments with special cases. I am really curious what the best practice of Cherry will turn out to be.

Ps. I mention for completeness and to conclude the thought, that any (`compile-string` generated) JS-code  can be eval'd in HTML and that calls whatever is in scope, which can e.g. be some foreign JS-ES6-module (that can in turn be another compiled Shadow-app).
